interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java

```diff
@@ -364,31 +375,43 @@ public void onNext(SimpleResponse response) {
                   }
                 }
               });
+          rpcsStarted.getAndIncrement();
         }
       }
-    }
 
-    long nanosPerQuery = TimeUnit.SECONDS.toNanos(1) / qps;
-
-    for (RpcType rpcType : rpcTypes) {
-      ListenableScheduledFuture<?> future =
-          exec.scheduleAtFixedRate(
-              new PeriodicRpc(rpcType), 0, nanosPerQuery, TimeUnit.NANOSECONDS);
+      private void handleRpcCompleted(long requestId, RpcType rpcType, String hostname,
+          Set<XdsStatsWatcher> watchers) {
+        int count = 0;
+        if (rpcsSucceededByPeer.containsKey(hostname)) {
+          count = rpcsSucceededByPeer.get(hostname);
+        }
+        rpcsSucceededByPeer.put(hostname, count + 1);
```

Comment:
Did I miss some synchronization barrier here? It seems like multiple calls to handleRpcCompleted could be executing simultaneously and `count` might no longer be correct by the time this `#put()` is invoked.

Reply:
You are right, this needs to be fixed. 

Btw, now that we are warming up the test client normally before making calls open for testing. So we can still use the existing way for checking if there is at least one RPCs succeeding on each backend (getting the stats for next `n` RPCs). Then we do not need to categorize the number of successful RPCs for each peer, we just need to know the total successful RPCs.

Reply:
Updated to count the total number of RPCs succeeded without counting them separately for each remote peer.